nservicebus/messaging/message-headers-for-third-party-integrations.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Required Headers For Third Party Integration
+summary: Lists the necessary headers when receiving messages from non-NSB endpoints for better monitoring.
+tags:
+- Monitoring
+- Third Party Integration
+- ServiceInsight
+- ServicePulse
+related:
+- nservicebus/messaging/message-headers
+---
+
+# How to integrate external systems with NServiceBus?
+
+When an NServiceBus endpoint receives messages from external systems such as BizTalk, TIBCO, etc directly, the message itself will not contain regular NServiceBus headers. Some headers are required by NServiceBus to handle those messages (i.e. enable deserialization of the message). Others are essential to provide effective monitoring and troubleshooting experience. 
+
+When NServiceBus endpoints send messages, they include a certain set of headers such as the type of message, the name of the sending endpoint, the machine name etc. To get the best use of your monitoring and debugging tools, please ensure that the following headers are being added to the actual message by the third party endpoints when sending a message. 
+
+### Mandatory Headers
+
+Required for deserializing the message:
+
+Key  | Value
+------------- | -------------
+NServiceBus.EnclosedMessageTypes  | Fully Qualified Message type including the assembly name, for e.g. IntegrationSample.Messages.Commands.ProcessOrder, IntegrationSample.Messages
```

Comment:
I think we need to check how older versions behave because this one isn't needed currently (in fact no headers are needed) -  see:

http://docs.particular.net/samples/sqltransport/native-integration/#code-walk-through-serialization (json)

and

http://docs.particular.net/samples/rabbitmq/native-integration/#code-walk-through-message-serialization (xml)